docs/changes.rst

```diff
@@ -22,6 +22,26 @@ You should generally assume that an API is internal unless you have specific
 information to the contrary.
 
 -------------------
+3.12.0 - 2017-06-21
+-------------------
+
+Hypothesis now understands inline type annotations (:issue:`293`):
+
+- If the first argument to :func:`~hypothesis.strategies.builds` has type
+  annotations, a default strategy for missing required arguments is selected
+  based on the type.  Type-based strategy selection will override a default
+  argument only if you explicitly pass :const:`~hypothesis.strategies.infer`.
+
+- When :func:`@given <hypothesis.given>` wraps a function with type
+  annotations, the wrapped test will have a default of
+  :const:`~hypothesis.strategies.infer` for arguments with a usable type hint.
```

Comment:
I would prefer `infer` to be a constant imported from `hypothesis` rather than `hypothesis.strategies` and leave the latter reserved entirely for strategies.